src/main/java/bhs/devilbotz/commands/arm/ArmSafety.java

```diff
@@ -167,5 +172,22 @@ public final boolean isFinished() {
    * execution of the command
    */
   public void initializeWithSafety() {}
-  ;
+
+  private boolean isArmStuck() {
```

Comment:
Instead of looking at the currentSpeed, I was thinking that we could look at if the position has changed since the last execution.  So, we'd have to remember the previousPosition and then if the position is the same for X calls in a row, then we assume the arm is stuck.

Reply:
Is there a reason for this? They should do the same thing

Reply:
I agree both ways should work.  In the end, we expect a the position to change by some _delta_ on every cycle.  We can detect it using position or velocity....we just need to calibrate the values accordingly.

Using currentSpeed requires us to know the set/expected speed to determine the "stuck" position.  Fortunately, the way the _current_ arm is physically designed, the speed is relatively constant.  _We should log the actual returned velocity returned by the encoder_ to see the range of values that are returned and set the "minSpeed" value accordingly.   FYI, if we ever decide to implement #127, the setSpeed needs to be taken into account, also.

In the end, we just need to detect the "stuck" condition _as fast as possible_ to prevent damage to the arm.

Reply:
Maybe can check two things.. if the motor speed is not zero and encoder distance is not changing, the arm is stuck.

Reply:
speed seems to work fine (as expected). @joshuman had to tweak the values to match the actual values reported from the encoder.getRate().  The typical rate is ~125-175, so we set the "stick" value to 75.  The code does seem to work in testing. When the arm is moving up, manually stopping the arm triggers the code correctly.  When the arm is moving down, the code does eventually trigger, but by the time it stops the arm, there is already too much tension on the arm.  We can maybe tweak tweak the values, but nonetheless, the code as-is is better than not having anything.